### PR TITLE
2 New peppers! Ghost and Cosmic Revenant!

### PR DESCRIPTION
**Pull request by jimmy12or**
## About the PR
This PR Adds 2 new Peppers! One is a Ghost Pepper that's so spicy it hurts, and the other is the Cosmic Revenant Pepper that produces Ectoplasm for Epistemics.

## Why / Balance
Botany still needs more unique plants.
The Ghost pepper is meant to inspire more food recipes primarily and to work as an in-between for the Cosmic Revenant.
The Cosmic Revenant is a Revenant inspired Pepper that is intended to be difficult to get due to producing Ectoplasm for Epistemics.

## Technical details
Rather basic code to add 2 new plants, one with animated sprites for the dead, Harvest, and produce.
Mutation line goes  Chili > Ghost Pepper > Cosmic Revenant. 
Assuming you don't get the Chilly Pepper.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
Screenshot 2024-07-26 041233
Screenshot 2024-07-26 041751
Screenshot 2024-07-26 041819
Screenshot 2024-07-26 041308

https://github.com/user-attachments/assets/789602ba-4d51-4e90-974b-3b8172c0b430

https://github.com/user-attachments/assets/0f1fb96d-94e3-4e42-bd81-37031988161a

https://github.com/user-attachments/assets/b156790a-c5a3-41b4-8623-6ece09aeacfe

## Breaking changes

None that I am aware of

**Changelog**

:cl:
- add: Added a Ghost Pepper! A friendly looking Pepper that's so Spicy, it hurts!
- add: Added a Cosmic Revenant! A spooky mutation of the Ghost pepper that produced Ectoplasm.

### CODE REVIEW

**Comment by deltanedas** [comment]
infinite ectoplasm is just stupidly broken epi would get hundreds of drains

**Comment by jimmy12or** [comment]
> infinite ectoplasm is just stupidly broken epi would get hundreds of drains

When showing off CrystalThistle, I was asked to make a plant that did Ectoplasm, which I had also been considering before. Revs are a royal pain and often just grief by spacing Evac and/or the entire station which is honestly a boring way to end a round. This gives Epi a potential way to help prevent Revs from happening in the first place, and since it's a plant that needs 2 mutations along with the first mutation competing with another, it's not going to be an easy one to get every round unless the botanist is really skilled or lucky at their job.

**Comment by deltanedas** [comment]
you can say that but omnizine is mass produced basically every round, ectoplasm would be the same. might as well remove the 500 glimmer objective or youd have to bomb botany roundstart

**Comment by DangerRevolution** [comment]
+1 for removing 500 glimmer

**Comment by jimmy12or** [comment]
> you can say that but omnizine is mass produced basically every round, ectoplasm would be the same. might as well remove the 500 glimmer objective or youd have to bomb botany roundstart

Using Omni as the example here works well. Omnizine is easy to get since it takes only one plant mutation on a plant that has no other mutations on it's pool for the time being, and yet while it is obtained most rounds, it isn't made every round. The chili plant has 2 options for mutation with this added. It has a 50:50 chance of going to Chilly peppers or ghost pepper. Then you have to get lucky or try very carefully to get it to the Cosmic Rev stage without the plant dying, producing a hazardous gas, or Kudzu'ing, this on top of the same hazards while making the ghost peppers before it. 

It's closer in difficulty to Spaceman's trumpets which are most certainly not grown every round, and that's again still without any other mutational branching since Poppies only go to Lily then Trumpet.

**Comment by Colin-Tel** [comment]
> Revs are a royal pain and often just grief by spacing Evac and/or the entire station which is honestly a boring way to end a round. 

To be honest with you, I would suggest that looking at what spawns a revenant and adjusting it so it isn't as often would be preferable compared to something that would alter the balance of glimmer as a whole.

**Comment by jimmy12or** [comment]
> > Revs are a royal pain and often just grief by spacing Evac and/or the entire station which is honestly a boring way to end a round.
> 
> To be honest with you, I would suggest that looking at what spawns a revenant and adjusting it so it isn't as often would be preferable compared to something that would alter the balance of glimmer as a whole.

Because this isn't only about Revs really. It is to provide another potential use/"customer" for Botany along with allowing for the option of more stability. Almost every shift while playing Epi, I've had to either kill a genuinely GOOD anomaly or stop all research just because the glimmer had gotten close to being too high. Being able to lower glimmer isn't a bad thing, and again, it's not an easy thing to grow.

**Comment by Colin-Tel** [comment]
The fact that it'd be possible to make 2 normality crystals with one Cosmic Revenant crop, seeing as it has a max ectoplasm of 10 per crop, just doesn't sit well with me.

File: Resources/Prototypes/DeltaV/Entities/Objects/Consumable/Food/produce.yml [hunk 1]
```diff
@@ -22,3 +22,69 @@
...
+          maxVol: 46
+          reagents:
+            - ReagentId: Nutriment
+              Quantity: 8
+            - ReagentId: CapsaicinOil
+              Quantity: 20
+            - ReagentId: Vitamin
+              Quantity: 8
+            - ReagentId: Ectoplasm
+              Quantity: 10
```
Comment (Colin-Tel) [inline Resources/Prototypes/DeltaV/Entities/Objects/Consumable/Food/produce.yml]:
```suggestion
              Quantity: 5
```

File: Resources/Prototypes/DeltaV/Hydroponics/seeds.yml [hunk 2]
```diff
@@ -23,3 +23,75 @@
...
+      Min: 4
+      Max: 10
+      PotencyDivisor: 25
+    Vitamin:
+      Min: 4
+      Max: 10
+      PotencyDivisor: 25
+    Ectoplasm:
+      Min: 1
+      Max: 10
```
Comment (Colin-Tel) [inline Resources/Prototypes/DeltaV/Hydroponics/seeds.yml]:
```suggestion
      Max: 5
```

**Review by Colin-Tel — COMMENTED** [review]
I would be fine with Ectoplasm being present with the following changes made.

**Comment by jimmy12or** [comment]
> I would be fine with Ectoplasm being present with the following changes made.

Before I make changes, I would like to at least state that a fully robusted Cosmic Revenant pepper only actually produces 6u, likely due to the fact that 100 potency is nearly impossible and robust harvest only brings it up to 50.

I can change how much is in the spawnable version but I don't personally feel lowering the amount in the growable is needed.

File: Resources/Prototypes/DeltaV/Hydroponics/seeds.yml (hunk 2 above)
Comment (jimmy12or) [reply Resources/Prototypes/DeltaV/Hydroponics/seeds.yml]: After testing, this still works well enough.

File: Resources/Prototypes/DeltaV/Entities/Objects/Consumable/Food/produce.yml (hunk 1 above)
Comment (jimmy12or) [reply Resources/Prototypes/DeltaV/Entities/Objects/Consumable/Food/produce.yml]: sure thing.

File: Resources/Textures/DeltaV/Objects/Specific/Hydroponics/Cosmic_Revenant.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1,87 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Taken from https://github.com/vgstation-coders/vgstation13, editted by Alchebun",
```
Comment (Colin-Tel) [inline Resources/Textures/DeltaV/Objects/Specific/Hydroponics/Cosmic_Revenant.rsi/meta.json]:
```suggestion
  "copyright": "Taken from https://github.com/vgstation-coders/vgstation13, edited by Alchebun",
```
:trollface:

**Review by Colin-Tel — APPROVED** [review]

File: Resources/Textures/DeltaV/Objects/Specific/Hydroponics/Cosmic_Revenant.rsi/meta.json (hunk 3 above)
Comment (jimmy12or) [reply Resources/Textures/DeltaV/Objects/Specific/Hydroponics/Cosmic_Revenant.rsi/meta.json]: fixed. I know you know that while spelling matters a lot in coding, this isn't one of those times. update named accordingly.

**Review by Colin-Tel — CHANGES_REQUESTED** [review]
untouch RT lol

File: RobustToolbox
Comment (DangerRevolution) [inline RobustToolbox:1]: F

**Comment by jimmy12or** [comment]
> untouch RT lol

I only removed the unneeded t from the word, what the heck happened here? Do you know what happened?

**Comment by Colin-Tel** [comment]
> I only removed the unneeded t from the word, what the heck happened here? Do you know what happened?

May want to revert your last commit, because it looks like you brought a submodule change onto the PR.

**Comment by jimmy12or** [comment]
> > I only removed the unneeded t from the word, what the heck happened here? Do you know what happened?
> 
> May want to revert your last commit, because it looks like you brought a submodule change onto the PR.

Still learning Git @_@

**Comment by Colin-Tel** [comment]
image
try hitting this thing instead

**Comment by jimmy12or** [comment]
> image try hitting this thing instead

Honestly, was worried that would make the CRLF check fail due to code being changed in Github rather than a standard commit. Glad that it doesn't.

**Review by Colin-Tel — APPROVED** [review]

**Comment by HandsomeTheEgg** [comment]
hey there , i know pr is closed but just wanted to leave some input , really liked the sprites and animation and ...dont understand the glimmer and ectoplasma stuff...but speaking of ectoplasma...another good usage would be if the ghost burger wouldnt require the item ectoplasma but like 5u of ectoplasma liquid , that way it also has another purpose instead of just epi , maybe for a future thing ( if the ectoplasma thing would be to hard you could also just make it like omnizine or koibeans 1-3u would tweak it a bit futher

**Comment by jimmy12or** [comment]
> hey there , i know pr is closed but just wanted to leave some input , really liked the sprites and animation and ...dont understand the glimmer and ectoplasma stuff...but speaking of ectoplasma...another good usage would be if the ghost burger wouldnt require the item ectoplasma but like 5u of ectoplasma liquid , that way it also has another purpose instead of just epi , maybe for a future thing ( if the ectoplasma thing would be to hard you could also just make it like omnizine or koibeans 1-3u would tweak it a bit futher

I feel this should be a pretty easy and straightforward change to make, I may see about doing that in the near future as a new PR.

**Pull request merged**